### PR TITLE
[Merlin Entertainments GB] New rulesets

### PR DESCRIPTION
**Pull request by youdly**
Related to https://github.com/EFForg/https-everywhere/pull/4003.

### CODE REVIEW

File: src/chrome/content/rules/AltonTowers.xml [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
-  
-  <securecookie host="^(?:.+\.)?altontowers\.com$" name=".*"/>
-  
-  <rule from="^http://(?:www\.)?altontowers\.com/" to="https://www.altontowers.com/"/>
+<!--
+	For other Merlin Entertainments coverage, see Merlin-Entertainments.xml.
+-->
+<ruleset name="Alton Towers Resort">
+	<target host="altontowers.com" />
+	<target host="www.altontowers.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/AltonTowers.xml]: WFM:
secure.altontowers.com

Timeout:
www2.altontowers.com

File: src/chrome/content/rules/FallsCreek.com.au.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	Mixed content:
+		- Font from fonts.gstatic.com *
+		- Font from fast.fonts.com *
+		- Image from chuckasnowie.com.au
+
+	* Secured by us
+-->
+<ruleset name="Falls Creek Resort" platform="mixedcontent">
+	<target host="fallscreek.com.au" />
+	<target host="www.fallscreek.com.au" />
```
Comment (J0WI) [inline src/chrome/content/rules/FallsCreek.com.au.xml]: Mismatch:
masterplan.fallscreek.com.au

File: src/chrome/content/rules/Legolanddiscoverycenter.xml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+<!--
+	For other Merlin Entertainments coverage, see Merlin-Entertainments.xml.
+-->
+<ruleset name="LEGOLAND Discovery Centre">
+	<target host="legolanddiscoverycenter.com" />
+	<target host="www.legolanddiscoverycenter.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/Legolanddiscoverycenter.xml]: WFM:
secure.legolanddiscoverycenter.com

Mismatch:
assets.legolanddiscoverycenter.com
app.legolanddiscoverycenter.com

File: src/chrome/content/rules/Legolanddiscoverycenter.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	
+	<target host="legolanddiscoverycentre.ca" />
+	<target host="www.legolanddiscoverycentre.ca" />
+
+	<target host="legolanddiscoverycentre.de" />
+	<target host="secure.legolanddiscoverycentre.de" />
+	<target host="www.legolanddiscoverycentre.de" />
+
+	<target host="legolanddiscoverycenter.cn" />
+	<target host="www.legolanddiscoverycenter.cn" />
```
Comment (J0WI) [inline src/chrome/content/rules/Legolanddiscoverycenter.xml]: WFM: shanghai.legolanddiscoverycenter.cn

File: src/chrome/content/rules/Madame-Tussauds.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<!--
+	For other Merlin Entertainments coverage, see Merlin-Entertainments.xml.
+
+	Nonfunctional subdomain:
+		- !www			(cert only matches www)
+-->
+<ruleset name="Madame Tussauds">
```
Comment (J0WI) [inline src/chrome/content/rules/Madame-Tussauds.xml]: WFM:
m.madametussauds.com
starwars.madametussauds.com
shanghai.madametussauds.com
cms.madametussauds.com

Refused:
assets.madametussauds.com

Mismatch:
reservations.madametussauds.com
(www.)promotix.madametussauds.com
justinbieber.madametussauds.com

Timeout:
beijing.madametussauds.com
gaga.madametussauds.com

File: src/chrome/content/rules/Merlin-Entertainments.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+		- The-Dungeons.xml
+		- Thorpe-Park.xml
+		- UnderwaterWorld.com.au.xml
+		- visitsealife.com.xml
+		- Warwick-Castle.xml
+		- Wildlifesydney.com.au.xml
+
+	Non-functional domain:
+		- merlinentertainments.biz		(timeout)
+		- merlincareers.com			(connection refused)
```
Comment (J0WI) [inline src/chrome/content/rules/Merlin-Entertainments.xml]: Same for www on both.
(www.)merlinentertainmentsgroup.com also times out

File: src/chrome/content/rules/The-Dungeons.xml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+		- Js from platform.twitter.com *
+
+	* Secured by us
+
+	Problematic subdomain:
+		- !www		(hostname mismatch, CN: *.azurewebsites.net)
+-->
+<ruleset name="The Dungeons">
+	<target host="thedungeons.com" />
+	<target host="www.thedungeons.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/The-Dungeons.xml]: WFM:
sanfrancisco.thedungeons.com
secure.thedungeons.com

File: src/chrome/content/rules/Thorpe-Park.xml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+<!--
+	For other Merlin Entertainments coverage, see Merlin-Entertainments.xml.
+
+	Nonfunctional domain:
+		- $self				(timeout)
+		- thorpeparkjobs.com		(connection refused)
+-->
+<ruleset name="Thorpe Park">
+	<target host="thorpepark.com" />
+	<target host="www.thorpepark.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/Thorpe-Park.xml]: WFM:
p12.thorpepark.com
secure.thorpepark.com

File: src/chrome/content/rules/Warwick-Castle.xml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+<!--
+	For other Merlin Entertainments coverage, see Merlin-Entertainments.xml.
+-->
+<ruleset name="Warwick Castle">
+	<target host="warwick-castle.com" />
+	<target host="www.warwick-castle.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/Warwick-Castle.xml]: WFM:
secure.warwick-castle.com

File: src/chrome/content/rules/visitsealife.com.xml [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<!--
+	For other Merlin Entertainments coverage, see Merlin-Entertainments.xml.
+-->
+<ruleset name="SEA LIFE">
+	<target host="visitsealife.com" />
+	<target host="www.visitsealife.com" />
+	<target host="www2.visitsealife.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/visitsealife.com.xml]: WFM:
secure.visitsealife.com

Timeout:
assets.visitsealife.com

**Comment by J0WI** [comment]
Posted some more subdomains.
Rules lgtm.

**Comment by J0WI** [comment]
@youdly are you still working on this?

File: src/chrome/content/rules/Thorpe-Park.xml (hunk 8 above)
Comment (youdly) [inline src/chrome/content/rules/Thorpe-Park.xml]: p12 - could not resolve host
secure - mismatch

**Comment by youdly** [comment]
Added subdomains and comments on non-functional subdomains.

File: src/chrome/content/rules/London-Eye.xml [hunk 11]
```diff
@@ -0,0 +1,17 @@
+<!--
+	For other Merlin Entertainments coverage, see Merlin-Entertainments.xml.
+-->
+<ruleset name="London Eye">
+	<target host="londoneye.com" />
+	<target host="secure.londoneye.com" />
```
Comment (J0WI) [inline src/chrome/content/rules/London-Eye.xml]: This times out now (see failing tests).

**Comment by youdly** [comment]
`secure.visitsealife.com` was fine in build [136595933](https://travis-ci.org/EFForg/https-everywhere/builds/136595933) but [136860699](https://travis-ci.org/EFForg/https-everywhere/jobs/136860699) gave `(35, 'gnutls_handshake() failed: The TLS connection was non-properly terminated.')`

**Comment by J0WI** [comment]
I have no idea what's wrong with Travis here.
Maybe @fuglede or @Hainish can help?

**Comment by fuglede** [comment]
Not sure; I reran the tests, and at least the first one went through.

**Comment by Hainish** [comment]
Not sure, worked on my local as well.  Re-running.

**Comment by fuglede** [comment]
This one is also pretty weird:

    ERROR src/chrome/content/rules/Warwick-Castle.xml: Fetch error: http://warwick-castle.com/ => https://warwick-castle.com/: (51, "SSL: certificate subject name (*.azurewebsites.net) does not match target host name 'warwick-castle.com'")

**Comment by J0WI** [comment]
@youdly to workaround these errors you can add the affected rulesets on `utils/ruleset-coverage-whitelist.txt`.

**Comment by youdly** [comment]
Whitelisted the affected ruleset `visitsealife.com.xml`. The error still persists though.

**Comment by fuglede** [comment]
If the only error is in the right half of the Travis test, it shouldn't be necessary to whitelist; a false negative should not carry over to a broken build post-merge.

**Comment by J0WI** [comment]
@fuglede would merging this breaking our build?

**Comment by fuglede** [comment]
@j0wi: The test works by comparing with the current `master` branch, so after putting the thing itself on master, the difference should be trivial, and the test not run. Be aware, though, that the PR fetch test is the same one which is run for the ruleset database wide fetch test, so if anybody runs that with the test set to disable erronous rulesets, then this will be caught in that process.

**Comment by Hainish** [comment]
Looks to be passing tests now.  Merged.

**Pull request merged**